src/groupByKey.ts

```diff
@@ -2,6 +2,17 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey(items: object[], key: string):
+GroupsMap<{[key: string]: object[]}> {
+  const groupedObject: GroupsMap<{[key: string]: object[]}> = {};
```

Comment:
```suggestion
GroupsMap<object> {
  const groupedObject: GroupsMap<{[key: string]: object[]}> = {};
```
You can use it like that. In type GroupsMap already mentioned the structure of object with keys and values like that.

Reply:
no, just use generic